src/store/modules/ModDownloadModule.ts

```diff
@@ -0,0 +1,112 @@
+import {ActionTree, GetterTree} from 'vuex';
+
+import {ImmutableProfile} from '../../model/Profile';
+import ThunderstoreCombo from '../../model/ThunderstoreCombo';
+import ThunderstoreDownloaderProvider from '../../providers/ror2/downloading/ThunderstoreDownloaderProvider';
+import {State as RootState} from '../index';
+import R2Error from "../../model/errors/R2Error";
+
+// TODO: Replace the "enum" in '../../model/enums/StatusEnum' with a proper enum like this one.
+enum StatusEnum {
+    FAILURE = 0,
+    PENDING = 1,
+    SUCCESS = 2,
+}
+
+interface ProgressItem {
+    modName: string;
+    installProgress: number;
+    downloadProgress: number;
+    status: number; // StatusEnum
+    error: R2Error | null;
+}
+
+interface State {
+    downloads: ProgressItem[];
+}
+
+/**
+ * State for Mod Downloading/Installation.
+ */
+export default {
+    namespaced: true,
+
+    state: (): State => ({
+        downloads: [],
+    }),
+
+    getters: <GetterTree<State, RootState>>{
+        activeDownload(state): ProgressItem | undefined {
+            return state.downloads.slice(-1)[0]; // Last element of the array
+        },
+    },
+    mutations: {
+        reset(state: State) {
+            state.downloads = [];
+        },
+        addDownload(state: State, modName: string) {
+            state.downloads.push({
+                modName: modName,
+                installProgress: 0,
+                downloadProgress: 0,
+                status: StatusEnum.PENDING,
+                error: null
+            });
+        },
+        updateDownloadProgress(state: State, params: { progress: number, modName: string, status: number, err: R2Error | null }) {
+            let downloadMod = state.downloads.find((progressItem) => progressItem.modName === params.modName);
+
+            // Doesn't do anything when the downloaded mod is not found in the array (which means it is a dependency).
+            // TODO: figure out if/how/where to track/display these dependencies' progresses.
+            // Might be affected by removing the completedCallback as one way to do it is to have it return
+            // a list of all of the downloaded mods (including the dependencies).
+
+            if (downloadMod) {
+                downloadMod.downloadProgress = params.progress;
+                downloadMod.status = params.status;
+                if (params.status === StatusEnum.FAILURE && params.err) {
+                    // TODO: do something with these failed downloads, i.e. display the error(s) to the user.
+                    downloadMod.error = params.err;
+                }
+            }
+        }
+    },
+    actions: <ActionTree<State, RootState>>{
+
+        //TODO: Do the installation
+        async downloadAndInstallMod(
+            {dispatch, getters, state, commit},
+            params: {
+                profile: ImmutableProfile,
+                mod: ThunderstoreCombo,
+            }
+        ) {
+            state.downloads.push({
```

Comment:
As per an earlier review comment, don't mutate state in the actions, use commits instead. In fact you have a mutator for this exact thing, but then it's not used for some reason.